api_utils/botEdit.py

```
@@ -0,0 +1,296 @@
+    "stub": ["لا لتخصيص البذرة"],
+    "tempcat": ["لا لإضافة صناديق تصفح معادلة"],
+    "portal": ["لا لربط البوابات المعادل", "لا لصيانة البوابات"],
+}
+
+
+def bot_May_Edit_do(text="", title_page="", botjob="all"):
+    # ---
+    """
+    Determines if a bot is permitted to edit a page based on templates in the page text.
+
+    Checks for the presence of templates that restrict bot editing, such as those in the stop lists for the specified bot job or the global list. Handles special cases for `nobots` and `bots` templates, interpreting their parameters to allow or deny editing for specific bots or all bots. Results are cached for efficiency.
```

Pylint (3.3.7)

[convention] Line too long (323/100)

(C0301)